vispy/visuals/markers.py

```diff
@@ -573,6 +574,8 @@ def __init__(self, scaling="fixed", alpha=1, antialias=1, spherical=False,
                           blend_func=('src_alpha', 'one_minus_src_alpha'))
         self._draw_mode = 'points'
 
+        self.events.add(data_updated=Event)
```

Comment:
Should we add this to the base visual to ensure it always exists? @djhoese

Reply:
I wouldn't be against that. I'd be a little worried that users seeing it would assume that it is properly emitted/handled by all visuals which won't be true after a lot of work. Updating every `set_data`, right? Regardless, moving it to the base classes and documenting that it is new in version X and may not be used/emitted by all Visuals would be a good start.

Reply:
Yeah, definitely document it. But I guess it's also not *too* much work to simply emit this in every visual's set_data.

Reply:
The problem becomes the Visuals that have other properties for setting data like `.color = new_color` and don't end up calling a `.set_data` method at all.

Reply:
If it's okay I would just leave it as-is for now. I can look into moving this to the base visual (and updating visuals to emit it) in another PR.

Reply:
Sounds good to me!